src/socket.rs

```diff
@@ -369,6 +369,49 @@ impl Socket {
         self.inner.set_ttl(ttl)
     }
 
+    /// Gets the value for the `SO_MARK` option on this socket.
+    ///
+    /// This value gets the socket mark field for each packet sent through
+    /// this socket.
+    ///
+    /// This function is only available on Unix and requires the
+    /// `CAP_NET_ADMIN` capability.
+    #[cfg(target_os = "linux")]
```

Comment:
Hey. I'm not a maintainer so I can't really approve your PR. However, this crate also uses the `all` feature as an opt-in to platform specific functions.

So this should probably be:
```suggestion
    #[cfg(all(feature = "all", target_os = "linux"))]
```

You may want to apply that to all the new functions of your PR.

And, as a user of the crate, thank you for the contribution :)

Reply:
As the maintainer, see above.

Reply:
It looks like v0.3.x doesn't use the `all` feature anywhere.